r/R/embedding.R

```diff
@@ -83,3 +61,67 @@ runEmbedding <- function(req, data) {
   res <- purrr::map2(df_embedding[[1]], df_embedding[[2]], map2_fun)
   return(res)
 }
+
+
+# getEmbedding
+# Return embedding calculated for the seurat object.
+# data is the seurat object.
+# config is the embedding work request config.
+# type is the embedding type.
+# reduction_type is the type of reduction that is used, e.g. PCA.
+# num_pcs is the number of principal components.
+#
+# Req$body has:
+#
+# Config has=
+# UMAP:
+# minimumDistance = float
+# distanceMetric = string (euclidean, cosine, etc)
+#
+# tsne:
+# perplexity
+# lerarningRate
+#
+#' @export
+getEmbedding <- function(config, method, reduction_type, num_pcs, data) {
+   if (method == "tsne") {
+    data <- Seurat::RunTSNE(data,
+      reduction = reduction_type,
+      seed.use = 1,
+      dims = 1:num_pcs,
+      perplexity = config$perplexity,
+      learning.rate = config$learningRate
+    )
+    result <- Seurat::Embeddings(data, reduction = method)
+  } else if (method == "umap") {
+    data <- Seurat::RunUMAP(data,
+      seed.use = 42,
```

Comment:
Could this be removed since we're already setting the seed [here](https://github.com/hms-dbmi-cellenics/worker/blob/f43024e96ddf76d9901a323b37b87591f2af8f73/r/R/embedding.R#L24)?